cmd/kube-scheduler/app/options/insecure_serving.go

```diff
@@ -98,50 +94,40 @@ func (o *CombinedInsecureServingOptions) ApplyToFromLoadedConfig(c *schedulerapp
 		return nil
 	}
 
-	if err := updateDeprecatedInsecureServingOptionsFromAddress(o.Healthz, componentConfig.HealthzBindAddress); err != nil {
-		return fmt.Errorf("invalid healthz address: %v", err)
-	}
-	if err := updateDeprecatedInsecureServingOptionsFromAddress(o.Metrics, componentConfig.MetricsBindAddress); err != nil {
-		return fmt.Errorf("invalid metrics address: %v", err)
-	}
+	updateDeprecatedInsecureServingOptionsFromAddress(o.Healthz, componentConfig.HealthzBindAddress)
+	updateDeprecatedInsecureServingOptionsFromAddress(o.Metrics, componentConfig.MetricsBindAddress)
 
 	return o.applyTo(c, componentConfig)
 }
 
-func updateAddressFromDeprecatedInsecureServingOptions(addr *string, is *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback) error {
+func updateAddressFromDeprecatedInsecureServingOptions(addr *string, is *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback) {
 	if is == nil {
 		*addr = ""
-	} else {
-		if is.Listener != nil {
-			*addr = is.Listener.Addr().String()
-		} else if is.BindPort == 0 {
-			*addr = ""
-		} else {
-			*addr = net.JoinHostPort(is.BindAddress.String(), strconv.Itoa(is.BindPort))
-		}
+		return
 	}
 
-	return nil
+	if is.Listener != nil {
+		*addr = is.Listener.Addr().String()
+	} else if is.BindPort == 0 {
+		*addr = ""
+	} else {
+		*addr = net.JoinHostPort(is.BindAddress.String(), strconv.Itoa(is.BindPort))
+	}
 }
 
-func updateDeprecatedInsecureServingOptionsFromAddress(is *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback, addr string) error {
+func updateDeprecatedInsecureServingOptionsFromAddress(is *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback, addr string) {
 	if is == nil {
-		return nil
+		return
 	}
+
 	if len(addr) == 0 {
 		is.BindPort = 0
-		return nil
-	}
-
-	host, portInt, err := splitHostIntPort(addr)
-	if err != nil {
-		return fmt.Errorf("invalid address %q", addr)
+	} else {
+		// In the previous `validate` process, we can ensure that the `addr` is legal, so ignore the error
```

Comment:
It's more future proof to leave the check

Reply:
Hi, thanks for your review!

In my eyes, there are some reasons why it could be revised in that way:

1.  First of all, the function is very simple and direct, just verifing and assigning. In my opinion, even in the future, the parameter `addr` should always be checked before using this function, so it is no problem to ignore the error returned by `splitHostIntPort` .
2.  Secondly, similar codes are common in this project. [For example](https://github.com/kubernetes/kubernetes/blob/24350a922ea2b1b2a8aeba58e150fa90370a282b/cmd/kube-proxy/app/server_others.go#L576-L577), there is no need to verify whether the `adr` is a legal IPv4 address before assigning a value to `nodes[0]`, cause it has been checked in the [validation](https://github.com/kubernetes/kubernetes/blob/24350a922ea2b1b2a8aeba58e150fa90370a282b/pkg/proxy/apis/config/validation/validation.go#L80-L101).
3.  Finally, this kind of processing could keep the relevant code logic more consistent, simpler and clearer.

Reply:
I don't have a strong opinion either way

/lgtm
/approve